### DISCUSSION TITLE
DataTable: support onClick on DataTableRow

### DISCUSSION THREAD

**Discussion by edoardo** [dhis2/notes#297] [Q&A]
Passing `onClick` on `DataTableRow` is not supported at the moment.
There are cases, like in the `OpenFileDialog` where it's desirable that the whole row is clickable, regardless of which cell the mouse hits.
I could obtain a similar result by adding the `onClick` on each cell of the row, but I don't see a reason why not have an optional `onClick` on the row component.

**Comment by Mohammer5** [comment]
I think this references the issue that we (it's one of our goals, right? @dhis2/team-platform) want to forward props to the root element / component

**Reply by HendrikThePendric** [reply]
Yeah, agreed. Ideally, I would like to replace our current `Table` components with the ones in `components/table/src/data-table/table-elements`, which already [forward props](https://github.com/dhis2/ui/blob/master/components/table/src/data-table/table-elements/table-row.js#L11).

**Comment by JoakimSM** [comment]
I second this request. Would it also make sense to use the pointer-cursor when an onClick event handler is passed? (goes for both the DataTableCell and DataTableRow)

**Reply by Mohammer5** [reply]
I think that depends on the implementation. If we add an explicit `onClick` prop, then yes, this would make a lot of sense.
Otherwise, if we just forward props, this should be done by providing a `className` or `style` prop

**Reply by edoardo** [reply]
Is this likely going to make it for 2.37?

**Comment by khalilurrrahmanridoykhan** [comment]
This is supported by the current component implementation.

`DataTableRow` collects unrecognised props in `...rest` and forwards them to the underlying `TableRow` ([current source](https://github.com/dhis2/ui/blob/fc2740b54a672583df936d1f53b25a1620aafde9/components/table/src/data-table/data-table-row/data-table-row.js#L11-L50)). An `onClick` handler can therefore be supplied directly:

```jsx
<DataTableRow
    onClick={() => openFile(file.id)}
    className="clickableRow"
>
    {/* cells */}
</DataTableRow>
```

For usability, add the pointer cursor only when the row is interactive. For accessibility, a row click alone is not enough: keep the primary action available as a real `button` or `a` element inside a cell so it is keyboard-focusable and has an accessible name. Also ensure buttons/links inside the row stop propagation if their action differs from the row action.

The generated API table may not list `onClick` specifically because it is forwarded as a native row prop, but the forwarding in the implementation is what enables it.
